pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/PersistentTopics.java

```diff
@@ -470,9 +474,22 @@ public PartitionedTopicMetadata getPartitionedMetadata(@PathParam("property") St
             @PathParam("destination") @Encoded String destination,
             @QueryParam("authoritative") @DefaultValue("false") boolean authoritative) {
         destination = decode(destination);
-       return getPartitionedTopicMetadata(property, cluster, namespace, destination, authoritative);
+        PartitionedTopicMetadata metadata = getPartitionedTopicMetadata(property, cluster, namespace, destination, authoritative);
+        if (metadata.partitions > 1) {
+            try {
+                validateClientVersion();
+            } catch (UnsupportedOperationException ue) {
```

Comment:
Would prefer isClientVersionValid instead of the throwing  and catching exceptions.

But otherwise - LGTM.

Reply:
yes, it make sense to not create exception twice. Probably we can do the similar as other validation-method which directly throws `RestException` with correct error-code and message. I will fix it.